Userland/head.cpp

```diff
@@ -1,25 +1,87 @@
+#include <errno.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <LibCore/CArgsParser.h>
+
+int head(String filename, bool print_filename, int line_count);
 
 int main(int argc, char** argv)
 {
-    // FIXME: Allow setting via command-line argument.
+    CArgsParser args_parser("head");
+
+    args_parser.add_arg("n", "lines", "Number of lines to print (default 10)");
+    args_parser.add_arg("q", "Never print filenames");
+    args_parser.add_arg("v", "Always print filenames");
+
+    CArgsParserResult args = args_parser.parse(argc, (const char**)argv);
+
     int line_count = 10;
+    if (args.is_present("n")) {
+        line_count = strtol(args.get("n").characters(), NULL, 10);
+        if (errno == EINVAL) {
```

Comment:
`strtol()` may also fail with `ERANGE` for values outside the `long` range.
Perhaps it would be better to check `if (errno)`